featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -507,10 +507,10 @@ def _calculate_transform_features(
         for f in features:
             # handle when no data
             if frame_empty:
-                set_default_column(frame, f)
-
+                feature_values.append(
+                    (f, [[] for _ in range(f.number_output_features)]),
```

Comment:
```suggestion
                    (f, [f.default_value for _ in range(f.number_output_features)]),
```

I think this would be better default handling